sway-lib-std/src/math.sw

```diff
@@ -97,13 +121,28 @@ impl Power for u256 {
 
         while exp > 1 {
             if (exp & 1) == 1 {
-                acc = acc * base;
+                // acc = acc * base;
+                let res = u256_checked_mul(acc, base);
+                acc = match res {
+                    Option::None => return u256::zero(),
```

Comment:
Why do you return `u256::zero()` here?

Reply:
Following the spec of u64. If u64 overflows on a pow operation it returns 0.